fvm/evm/emulator/emulator.go

```diff
@@ -279,3 +284,25 @@ func (proc *procedure) run(msg *gethCore.Message, txType uint8) (*types.Result,
 	}
 	return &res, err
 }
+
+func SetupPrecompile(cfg *Config) error {
+	rules := cfg.ChainRules()
+	// captures the pointer to the map that has to be augmented
+	var precompiles map[gethCommon.Address]gethVM.PrecompiledContract
+	switch {
+	case rules.IsCancun:
+		precompiles = gethVM.PrecompiledContractsCancun
+	case rules.IsBerlin:
+		precompiles = gethVM.PrecompiledContractsBerlin
+	case rules.IsIstanbul:
+		precompiles = gethVM.PrecompiledContractsIstanbul
+	case rules.IsByzantium:
+		precompiles = gethVM.PrecompiledContractsByzantium
+	default:
+		precompiles = gethVM.PrecompiledContractsHomestead
+	}
+	for k, v := range cfg.ExtraPrecompiles {
+		precompiles[k] = v
```

Comment:
Might be good to improve the naming of `k` and `v`, and maybe prevent overwrites